File: pydroid/src/main/java/com/pyamsoft/pydroid/about/AboutLibrariesModel.java
/*
 * Copyright 2017 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.pyamsoft.pydroid.about;

import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;
import android.support.annotation.RestrictTo;
import com.google.auto.value.AutoValue;

@RestrictTo(RestrictTo.Scope.LIBRARY) @AutoValue public abstract class AboutLibrariesModel {

  @CheckResult @NonNull
  public static AboutLibrariesModel create(@NonNull String name, @NonNull String homepage,
      @NonNull String license) {
    return new AutoValue_AboutLibrariesModel(name, homepage, license, "");
  }

  @CheckResult @NonNull
  public static AboutLibrariesModel createWithContent(@NonNull String name, @NonNull String homepage,
      @NonNull String content) {
    return new AutoValue_AboutLibrariesModel(name, homepage, "", content);
  }

  @CheckResult public abstract String name();

  @CheckResult public abstract String homepage();

  @CheckResult public abstract String license();

  @CheckResult public abstract String customContent();
}

File: pydroid/src/main/java/com/pyamsoft/pydroid/about/AboutLibrariesModel.kt
/*
 *     Copyright (C) 2017 Peter Kenji Yamanaka
 *
 *     This program is free software; you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation; either version 2 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License along
 *     with this program; if not, write to the Free Software Foundation, Inc.,
 *     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package com.pyamsoft.pydroid.about

import android.support.annotation.CheckResult

data class AboutLibrariesModel internal constructor(val name: String, val homepage: String,
    val license: String, val customContent: String) {

  companion object {

    @JvmStatic
    @CheckResult
    fun create(name: String, homepage: String,
        license: String): AboutLibrariesModel = AboutLibrariesModel(name, homepage, license, "")

    @JvmStatic
    @CheckResult
    fun createWithContent(name: String, homepage: String,
        content: String): AboutLibrariesModel = AboutLibrariesModel(name, homepage, "", content)
  }
}
